src/lib.rs

```diff
@@ -1,6 +1,560 @@
+use core::panic;
+
 #[doc = include_str!("../README.md")]
 
-/// Say hi to the entire world :)
-pub fn greet() {
-	println!("Hello World!");
+#[derive(PartialEq, Eq)]
+#[derive(Debug)]
+pub enum StringDiffOpKind {
+    Substitute(char, char),  
+    Insert(char),
+    Delete(char),
+    Transpose(usize, usize),
+}
+
+#[derive(Debug)]
+pub struct StringDiffOp {
+    pub kind: StringDiffOpKind,
+    pub index: isize,
```

Comment:
Is there a specific reason this is an `isize` (signed)? This would make more sense as a `usize` (unsigned) since you can't negatively index a string